File: sms-service/src/main/java/com/sms/service/utils/rabbit/converter/NotificationMessageConverter.java
package com.sms.service.utils.rabbit.converter;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.sms.service.entity.Notification;
import lombok.RequiredArgsConstructor;
import lombok.SneakyThrows;
import org.springframework.amqp.core.Message;
import org.springframework.amqp.core.MessageProperties;
import org.springframework.amqp.support.converter.AbstractMessageConverter;
import org.springframework.amqp.support.converter.MessageConversionException;
import org.springframework.stereotype.Component;

@Component("notificationMessageConverter")
@RequiredArgsConstructor
public class NotificationMessageConverter extends AbstractMessageConverter {
    private final ObjectMapper objectMapper;

    @SneakyThrows
    @Override
    protected Message createMessage(Object object, MessageProperties messageProperties) {
        byte[] payload = objectMapper.writeValueAsBytes(object);
        return new Message(payload, messageProperties);
    }

    @SneakyThrows
    @Override
    public Notification fromMessage(Message message) throws MessageConversionException {
        return objectMapper.readValue(message.getBody(), Notification.class);
    }
}

File: notification-service/src/main/kotlin/ua/notification/service/utils/broker/converter/NotificationMessageConverter.kt
package ua.notification.service.utils.broker.converter

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.amqp.core.Message
import org.springframework.amqp.core.MessageProperties
import org.springframework.amqp.support.converter.AbstractMessageConverter
import org.springframework.stereotype.Component
import ua.notification.service.entity.additional.notification.Notification

@Component
class NotificationMessageConverter(
    private val objectMapper: ObjectMapper
) : AbstractMessageConverter() {

    override fun fromMessage(message: Message): Notification {
        val payload = message.body
        return objectMapper.readValue(payload, Notification::class.java)
    }

    override fun createMessage(obj: Any, messageProperties: MessageProperties): Message {
        val payload = objectMapper.writeValueAsBytes(obj)
        return Message(payload, messageProperties)
    }
}